### ISSUE TITLE
Ubuntu support

### ISSUE THREAD

**Issue by kaovilai** [openshift/microshift#984]
We want to be able to start and run microshift from ubuntu which is used by many ci systems.

**Comment by ggiguash** [comment]
@kaovilai, we are not planning to support Operator Lifecycle Manager (OLM) with MicroShift.

Operators may still be installed if they do not require any OCP APIs unsupported by MicroShift. You should inquire with the OADP team if non-OLM installation is possible/supported.

**Comment by kaovilai** [comment]
@ggiguash  I am part of the OADP team and we are investigating ways to get around 40minute install time for e2e testing if possible. Microshift is one of the possible avenues to explore.

**Comment by kaovilai** [comment]
Looked into Hive cluster but it seems like we may not have enough permission there to install operators unless we make our own hive.

**Comment by ggiguash** [comment]
> @ggiguash I am part of the OADP team and we are investigating ways to get around 40minute install time for e2e testing if possible. Microshift is one of the possible avenues to explore.

Please, review the list of OCP API supported by MicroShift at https://github.com/openshift/microshift/blob/main/docs/enabled_apis.md. 
We have no plans to enable OLM support in the foreseeable future. 

Let's close this issue, unless there is any further questions on OADP installation without OLM.

**Comment by kaovilai** [comment]
We are not worried about OADP installation specifically. This issue is not regarding any operator installation or OLM. The ask is simply, can we have something like `kind create cluster` for microshift.

> If you have [go](https://golang.org/) ([1.17+](https://golang.org/doc/devel/release.html#policy)) and [docker](https://www.docker.com/) installed go install sigs.k8s.io/kind@v0.17.0 && kind create cluster is all you need!

https://kind.sigs.k8s.io/

**Comment by kaovilai** [comment]
I am not asking for additional API support, OLM support, OADP support from microshift.

I am asking that microshift be as easy to install and up and running ready for kubectl/oc like _kind_ cluster

**Comment by kaovilai** [comment]
specifically on non RHEL systems that are used by travis-ci/circle-ci if possible.

**Comment by kaovilai** [comment]
I hope title change clear up the minimal requirements.

**Comment by bshephar** [comment]
Hey,

I'm not sure if Ubuntu support makes a lot of sense for this project. Microshift is a project build for the Red Hat family of operating systems. Designed to be a light weight, edge worthy deployment of OpenShift and ideally run on os-tree bistro deployments.

In that regard, I'm not sure the mission of the Microshift project would include support for multiple Operating Systems. I understand what you're saying about upstream CI environments. But I don't think multi-OS support is something this project is pursuing. 

With regards to kind-like deployment that can be up and running quickly. Have you tried Code Ready Containers?
https://github.com/code-ready/crc
https://developers.redhat.com/blog/2019/09/05/red-hat-openshift-4-on-your-laptop-introducing-red-hat-codeready-containers

Maybe that would be better suited for your use case?

**Comment by ggiguash** [comment]
@kaovilai, At this time, we only plan supporting MicroShift on RHEL 8.x and, later, RHEL 9.x OS. Furthermore, due to the nature of the project, we are not planning to support container-based bootstrap for MicroShift. All these for downstream activities.

When upstream activities are resumed, it is likely that CentOS and Fedora are going to be used by the community.

Overall, the only configuration we can offer for running MicroShift on Ubunty is to bootstrap a RHEL VM and run MicroShift in it. An example of such a quick-starter can be found at https://github.com/ggiguash/microshift/blob/main/docs/getting_started.md.

**Comment by kaovilai** [comment]
CRC do not work.
https://app.circleci.com/pipelines/github/kaovilai/reliable-ci/15/workflows/e65e2e00-0e06-44da-9598-a0d1a0e593a3/jobs/22

**Comment by kaovilai** [comment]
Microshift had Ubuntu install scripts previously iiuc but later removed.

I just need a flavor of openshift that runs in Travis/circleci. Point me to one if you know anything that works ☺️.

**Comment by kaovilai** [comment]
Apart from specific distribution support.

kind cluster support any environment with a valid docker and golang install.

Not so microshift.

**Comment by ggiguash** [comment]
I'm afraid that's all we can offer at this time. There are no plans to support Ubuntu or containerized MicroShift deployment as part of the downstream activities.

**Comment by lbrigman124** [comment]
I did notice a project that is working with MicroShift in Docker - https://github.com/christianh814/fauxpenshift

**Comment by Ozfer** [comment]
Ubuntu/Debian support is still relevant. Many edge devices are not certified to run Fedora or it's just not possible without extensive hacking. There used to be support for ubuntu in this project last year. If part of this microshift effort is goodwill and openness for the community and driving growth of Openshift, this should be considered. https://community.ibm.com/community/user/cloud/blogs/alexei-karve/2022/04/17/microshift-13 
/remove-lifecycle stale

**Comment by kmf** [comment]
Agree @Ozfer it's very important. 
It's almost as foundational as Podman. 

@kaovilai have you tried k0s/k3s?

**Comment by kaovilai** [comment]
I've tried kind cluster which works well but lacks openshift apis

**Comment by ggiguash** [comment]
> I've tried kind cluster which works well but lacks openshift apis

@kaovilai , note that MicroShift does not support full set of OpenShift APIs.
See https://github.com/openshift/microshift/blob/main/docs/contributor/enabled_apis.md for more information.

**Comment by kaovilai** [comment]
Yeah that should be good enough.. and then I assume I can add operatorhub on top as necessary.

**Comment by kaovilai** [comment]
BTW, unblocking other OSes would be a huge help in getting more community contributions to fixing the project IMO.. and reduce likelihood of this project adoption being limited.. but perhaps that's by design. As seen in [LinkedIn post](https://www.linkedin.com/posts/cwayne18_k3s-activity-7112099145963745280-vn0V?utm_source=share&utm_medium=member_desktop)

## Star History

[![star-history-2023927](https://github.com/openshift/microshift/assets/11228024/8ae1c885-cbda-4139-ae3d-1ee852304d88)](https://star-history.com/#openshift/microshift&k0sproject/k0s&k3s-io/k3s&canonical/microk8s&Date)

**Comment by codespearhead** [comment]
I found this [Microshift installer script that doesn't require a pull secret](https://github.com/nerdalert/microshift-installer/blob/main/README.md). Perhaps that's a good starting point for an eventual PR that will solve this issue.

**Comment by ggiguash** [comment]
@codespearhead , the script installs a 2-year-old microshift package from `@redhat-et/microshift`.
See https://github.com/nerdalert/microshift-installer/blob/8cb53a452d0461555026a8d9c372242d7b0327fc/microshift-install.sh#L40

**Comment by ggiguash** [comment]
> In that case, which one the tags identify the latest version that should be published to @redhat-et/microshift?
We stopped publishing MicroShift to `@redhat-et/microshift` copr repository long time ago. It contains outdated s/w.

**Comment by DanielFroehlich** [comment]
/close
this is a rotten issue. Centos streams could be used, also podman desktop meanwhile has MicroShift integration. We meanwhile also publish bootc images that can be used to directly boot a MicroShift instance. We are also working on providing a pure OKD based deployment which will not need a pull secret.

**Comment by kaovilai** [comment]
Centos is not an option for circle CI if I'm remembering correctly. Running in a container will likely be difficult
